lib/websocket.ts

```diff
@@ -168,15 +178,18 @@ export class FireFlyWebSocket {
     }
   }
 
-  close() {
-    this.closed = true;
+  async close(wait?: boolean): Promise<void> {
+    const closedPromise = new Promise<void>(resolve => {
+      this.closed = resolve;
+    });
     this.clearPingTimers();
     if (this.socket) {
```

Comment:
Rather than relying on the other "close" handler and dealing with callbacks back and forth, would it be possible to just add another handler here?
```
const closedPromise = new Promise<void>(resolve => this.socket.on('close', resolve));
```

Reply:
I'm not sure I can for the use case I'm fighting. Jest cannot handle any logging after a test completes, so what I need to know is that the _existing_ callback that does logging has done it.

If I add another callback as you propose, then I'll be relying on the order of invocation of event handlers. Happy to do that if you'd prefer (I have written up the code for it ready to push, but I'll need to test it).

Reply:
Discussed in a message directly with @awrichar and deciding to merge.